src/main/java/com/fasterxml/jackson/databind/deser/DeserializerCache.java

```diff
@@ -34,15 +35,14 @@ public final class DeserializerCache
      * specifically, ones that are expensive to construct.
      * This currently means bean, Enum and container deserializers.
      */
-    final protected LRUMap<JavaType, JsonDeserializer<Object>> _cachedDeserializers;
+    final protected LookupCache<JavaType, JsonDeserializer<Object>> _cachedDeserializers;
 
     /**
      * During deserializer construction process we may need to keep track of partially
      * completed deserializers, to resolve cyclic dependencies. This is the
      * map used for storing deserializers before they are fully complete.
      */
-    final protected HashMap<JavaType, JsonDeserializer<Object>> _incompleteDeserializers
-        = new HashMap<JavaType, JsonDeserializer<Object>>(8);
+    final protected HashMap<JavaType, JsonDeserializer<Object>> _incompleteDeserializers;
```

Comment:
No, let's keep initialization here since we don't ever want to share/reuse map -- this is only used during initialization of individual `JsonDeserializer`s, to resolve cyclic dependencies.